testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -247,6 +248,22 @@ public void testRowLevelDelete()
         }
     }
 
+    @Test
+    public void testTruncateTable()
+    {
+        if (!hasBehavior(SUPPORTS_TRUNCATE)) {
+            assertQueryFails("TRUNCATE TABLE nation", "This connector does not support truncating tables");
```

Comment:
Please note that this logic has a risk to truncate a nation table in SaaS environment. I had to restore the table in BigQuery because `TestBigQueryArrowConnectorSmokeTest` didn't enable `SUPPORTS_TRUNCATE`.

Reply:
Thanks. I will modify it with the below code.

```java
try (TestTable table = new TestTable(getQueryRunner()::execute, "test_truncate", "AS SELECT * FROM nation")) {
    assertQueryFails("TRUNCATE TABLE " + table.getName(), "This connector does not support truncating tables");
}
```

should we modify here as well?

https://github.com/trinodb/trino/blob/0df147669c929fec6dabfff3287c8eea36b403f1/testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java#L4624-L4627